stacker/stack.py

```diff
@@ -84,6 +84,12 @@ def __repr__(self):
 
     @property
     def requires(self):
+        # By definition, a locked stack has no dependencies, because we won't
+        # be performing an update operation on the stack. This means, resolving
+        # outputs from dependencies is unnecessary.
+        if self.locked:
```

Comment:
This change is pretty useful by itself, even if we don't add the `--only` flag. There's no reason for locked stacks to depend on anything, because by definition, they don't depend on anything.

Reply:
Good catch. If the stack is in `--forced`, it still gets updated regardless of if it's locked or not:

https://github.com/cloudtools/stacker/blob/87688453c64ef333c48786000f2f890cbe664633/stacker/actions/build.py#L39-L57

It might be better if we moved that check to where we initialize the `stacker.Stack`, and just unlock the stack if it's in `--force`, so that `locked` always means what you think it means.

Reply:
Hmm, yeah, I kind of like the "cleanness" of the way it is now - that a stack is locked if it's locked, but that we ignore it if we're told to.  That said, I do like the feature you're building in here.

Another question - how would this affect things like the `dag` command?  If something was locked, would it look like it was an unattached node of the graph?